components/uniset/src/utils.rs

```diff
@@ -6,6 +6,8 @@ use core::{
     char,
     ops::{Bound::*, RangeBounds},
 };
+use zerovec::ule::AsULE;
+use zerovec::ZeroVec;
 
 /// Returns whether the vector is sorted ascending non inclusive, of even length,
 /// and within the bounds of `0x0 -> 0x10FFFF` inclusive.
```

Comment:
I think you don't need both versions of `is_valid`.  If you get a `&[u32]`, you can convert it to a `ZeroVec<u32>` before validating.